### PR TITLE
allow absolute path to entity-server persist file

### PR DESCRIPTION
**Pull request by birarda**
- added handling of absolute entity persist file path from domain-server settings

### CODE REVIEW

File: assignment-client/src/octree/OctreeServer.cpp [hunk 1]
```diff
@@ -1145,8 +1144,21 @@ void OctreeServer::domainSettingsRequestComplete() {
     if (_wantPersist) {
         // If persist filename does not exist, let's see if there is one beside the application binary
         // If there is, let's copy it over to our target persist directory
-        auto persistPath = ServerPathUtils::getDataFilePath("entities/" + QString(_persistFilename));
-        if (!QFile::exists(persistPath)) {
+        QDir persistPath { _persistFilePath };
+        QString absoluteFilePath = persistPath.path();
```
Comment (huffman) [inline assignment-client/src/octree/OctreeServer.cpp]: Did you want to use `persistPath.absolutePath()` here?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 90b78feb1ee27ecd3dd35ab31f453f4912362e5b available at: https://gist.github.com/b30fa16168eb86f1bd9a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.dmg

**Comment by birarda** [comment]
To test this PR you'll want to test both the migration of your previous path and then the use of an absolute persist path.

Before you run the domain-server from this branch for the first time, set a custom Entities File Path in your domain-server. Then, once the domain-server from this is run confirm that it is pointing at the same file.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2c1623ed427b326a844c276e663a4ed9d377ac1f available at: https://gist.github.com/b30fa16168eb86f1bd9a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.dmg

**Comment by huffman** [comment]
To test:
1. Open interface, go to 0,0,0
2. Go into your domain settings and add this to the persistent script list: https://gist.githubusercontent.com/huffman/0ec1c70ba7af1904af8a/raw/727c20297b452745a736633ebc363a9904a51e36/acSimTest.js - this script will create a box at 0,0,0 and move a sphere inside of it every 4 seconds.
3. From interface, observe the sphere, and after a few seconds, confirm that it moves. Confirm that after another 4 seconds it moves again. Previously the behavior was that the sphere would jump back to the center of the box.

**Comment by huffman** [comment]
Oops, commented on wrong pr

**Comment by OmegaHeron** [comment]
@birarda - Initial test is this = good.  Let me do one more set of tests then I'll comment again.

**Comment by nbq** [comment]
I also tested this and will say I setup a status page and if you enable download persist file, on the status page with a custom entity path, it shows up as "http://home/hifi/run/resources/models.json.gz" with url of "http://home/hifi/run/resources/models.json.gz"

**Comment by OmegaHeron** [comment]
Other than status page issue above - this works correctly for me.

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea9d84bdc47556b9e31dad5385b566840417723c available at: https://gist.github.com/b30fa16168eb86f1bd9a<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7019.dmg

**Pull request merged**